### PR TITLE
Context api store

### PR DESCRIPTION
**Pull request by qburst-supriya**
Added the following
               Store using Context API as generic
               Custom hook to update Context outside from Presentation Component
               Axios to fetch Rest APIs as generic
               Added API base url as env variable (Just for reference committed to Git. Will ignore this in next commit)
              Added prettier file
              Path alias has been configured. Yet to refer in imports

### CODE REVIEW

File: Bus Scheduler/src/App.css [hunk 1]
```diff
@@ -1,42 +1,42 @@
 #root {
-  max-width: 1280px;
-  margin: 0 auto;
-  padding: 2rem;
-  text-align: center;
+    max-width: 1280px;
+    margin: 0 auto;
+    padding: 2rem;
+    text-align: center;
 }
```
Comment (lekshmi615213) [inline Bus Scheduler/src/App.css:6]: better usage of scss

File: Bus Scheduler/src/components/tripSearch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useFetchTrips } from './store/hook';
+
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/index.tsx]: if this component is part of a page then  /features/home/components/tripSearch or else if it is shared /src/shared/components/tripSearch/

File: Bus Scheduler/src/components/tripSearch/store/hook.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { GET } from '../../../store/api/axios'
+import { tripListApiUrl } from '../../../store/api/urls'
+
+import { useCustomContext } from '../../../store/contextProvider';
+import { TripSearchContext } from './tripSearchProvider';
+
+export const useFetchTrips = () => {
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/store/hook.ts]: custom hooks can be /components/component-name/hooks

File: Bus Scheduler/src/components/tripSearch/store/hook.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    useEffect(() => {
+        const fetchTrips = async () => {
+            const response = await GET(tripListApiUrl);
+            return response;
+        }
+        const tripList = fetchTrips();
+
+        tripList.then((response) => {
+            actions.listTrips({ data: response });
+        })
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/store/hook.ts]: good to follow delcarative code, encapsulate the code in a function and use it in useEffect

File: Bus Scheduler/src/components/tripSearch/store/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { ActionType } from "../../../store/types";
+import { ActionKind, TripListType } from "./types";
+
+
+export const TripSearchReducer = (state: TripListType, action: ActionType<TripListType>): TripListType => {
+  switch (action.type) {
+    case ActionKind.LIST_TRIP: {
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/store/reducer.ts]: use context api ?

File: Bus Scheduler/src/components/tripSearch/store/tripSearchProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    const listTrips = (payload: TripListType) => {
+        dispatch({ type: ActionKind.LIST_TRIP, payload: { ...payload } });
+    };
+    const actions: TripSearchActionsType = { listTrips };
+    const contextvalue: BaseContextType<TripListType> = {
+        data,
+        actions,
+    };
+
+    const ContextObject = TripSearchContext as unknown as React.Context<BaseContextType<TripListType>>;
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/store/tripSearchProvider.tsx]: why aliasing?

File: Bus Scheduler/src/components/tripSearch/store/types.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+
+export type TripListType =
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/store/types.ts]: TripList is fine

File: Bus Scheduler/src/components/tripSearch/store/types.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  {
+    data: {
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/store/types.ts]: good to create a sub type and use it here

File: Bus Scheduler/src/components/tripSearch/store/types.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+export type TripListContextType = {
+  data: TripListType;
+  actions: TripSearchActionsType
+};
+
+export type TripSearchActionsType = {
+  listTrips: (payload: TripListType) => void
+}
+
+export enum ActionKind {
```
Comment (lekshmi615213) [inline Bus Scheduler/src/components/tripSearch/store/types.ts]: can we create a new file enum and move it there

File: Bus Scheduler/src/index.css [hunk 10]
```diff
@@ -1,68 +1,72 @@
...
-    background-color: #ffffff;
-  }
-  a:hover {
-    color: #747bff;
-  }
-  button {
-    background-color: #f9f9f9;
-  }
+    :root {
+        color: #213547;
```
Comment (lekshmi615213) [inline Bus Scheduler/src/index.css:63]: color, font etc can be variables

File: Bus Scheduler/src/store/api/axios.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
+import axios, { AxiosError } from "axios";
+import { baseApiUrl } from "./urls";
+
```
Comment (lekshmi615213) [inline Bus Scheduler/src/store/api/axios.ts]: can be services/api

File: Bus Scheduler/src/store/api/axios.ts [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    return config;
+  },
+  (error) => {
+    return Promise.reject(error);
+  },
+);
+
+
+// Response interceptor
+baseApi.interceptors.response.use(
```
Comment (lekshmi615213) [inline Bus Scheduler/src/store/api/axios.ts]: good to use seperate files for interceptors

File: Bus Scheduler/src/App.css (hunk 1 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/App.css:6]: Will incorporate while doing styling

File: Bus Scheduler/src/components/tripSearch/index.tsx (hunk 2 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/components/tripSearch/index.tsx]: Done

File: Bus Scheduler/src/components/tripSearch/store/hook.ts (hunk 3 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/components/tripSearch/store/hook.ts]: Done

File: Bus Scheduler/src/components/tripSearch/store/hook.ts (hunk 4 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/components/tripSearch/store/hook.ts]: Done

File: Bus Scheduler/src/components/tripSearch/store/reducer.ts (hunk 5 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/components/tripSearch/store/reducer.ts]: Done

File: Bus Scheduler/src/components/tripSearch/store/types.ts (hunk 8 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/components/tripSearch/store/types.ts]: Done

File: Bus Scheduler/src/components/tripSearch/store/types.ts (hunk 9 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/components/tripSearch/store/types.ts]: Done

File: Bus Scheduler/src/index.css (hunk 10 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/index.css:63]: Will incorporate while doing styling

File: Bus Scheduler/src/store/api/axios.ts (hunk 11 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/store/api/axios.ts]: Done

File: Bus Scheduler/src/store/api/axios.ts (hunk 12 above)
Comment (qburst-supriya) [reply Bus Scheduler/src/store/api/axios.ts]: Done

**Pull request merged**